helper/communicator/step_connect.go

```diff
@@ -44,9 +45,28 @@ type StepConnect struct {
 	substep multistep.Step
 }
 
+func (s *StepConnect) pause(pauseLen time.Duration, ctx context.Context) bool {
+	// Use a select to determine if we get cancelled during the wait
+	log.Printf("Pausing before connecting...")
+	select {
+	case <-ctx.Done():
+		return true
+	case <-time.After(pauseLen):
+	}
```

Comment:
Nice one !!!

Just as an example; you could also have done:

```go
ctx, _ = context.WithTimeout(ctx, pauseLen)
err := <- ctx.Done()
// ...
```

Reply:
ah, neato. I didn't know you could stack contexts like that.